src/stream.rs

```diff
@@ -0,0 +1,85 @@
+use crate::key::RedisKey;
+use crate::raw;
+use crate::RedisError;
+use crate::RedisString;
+use std::os::raw::c_long;
+use std::ptr;
+
+pub struct StreamRecord {
+    pub id: raw::RedisModuleStreamID,
+    pub fields: Vec<(RedisString, RedisString)>,
+}
+
+pub struct StreamIterator<'key> {
+    key: &'key RedisKey,
+}
+
+impl<'key> StreamIterator<'key> {
+    pub(crate) fn new(
+        key: &RedisKey,
+        mut from: Option<raw::RedisModuleStreamID>,
+        mut to: Option<raw::RedisModuleStreamID>,
+        exclusive: bool,
```

Comment:
@MeirShpilraien what about REDISMODULE_STREAM_ITERATOR_REVERSE?

Reply:
Good point, will add it.

Reply:
Done